src/brod_utils.erl

```diff
@@ -985,6 +986,50 @@ is_batch(_) -> false.
 nolink(C) when is_list(C) -> [{nolink, true} | C];
 nolink(C) when is_map(C) -> C#{nolink => true}.
 
+%% @doc Parallel map with timeout.
+-define(IS_TIMEOUT(T), ((is_integer(Timeout) andalso Timeout >= 0) orelse Timeout =:= infinity)).
+-spec pmap(fun((A) -> B), list(A), timeout()) -> list(B).
+pmap(_Fun, [], _Timeout) -> [];
+pmap(Fun, [_|_] = List, Timeout) when is_function(Fun, 1), ?IS_TIMEOUT(Timeout) ->
+  SeqL = lists:seq(1, length(List)),
+  Parent = self(),
+  Collector = erlang:spawn_link(fun() -> collect_pmap_results(Parent, SeqL, []) end),
+  Workers = lists:map(fun({Seq, Item}) ->
+                        erlang:spawn_link(fun() -> Collector ! {Seq, Fun(Item)} end)
+                      end, lists:zip(SeqL, List)),
+  receive
+    {Collector, Res} ->
+      ok = drain_exits([Collector | Workers]),
+      Res
+  after Timeout ->
+    lists:foreach(fun unlink_kill/1, [Collector | Workers]),
```

Comment:
So the following drain_exits are for the best efforts? as unlink then kill won't promise a EXIT message.